### PR TITLE
ACI resource: Add additional volume support. Refactor volume structure

### PR DESCRIPTION
**Pull request by lawrencegripper**
Resolves: #1860 

Currently working on refactoring the existing provider to support volume mounts and allow volumes to be shared between containers. 

Goal: Add missing volume types

### CODE REVIEW

**Comment by lawrencegripper** [comment]
Hi, 

I think I've got the basics of how I want to tackle this now in the PR - would someone be able to take a quick look and let me know their thoughts? 

If everything looks good I'll finish things off with the additional volume types and do docs, tests etc

**Comment by lawrencegripper** [comment]
Hopefully this is in a good state for an initial review now. 

This ended up larger than I expected as the API doesn't return 'secrets' or azure storage keys in the read requests so I've added quite a bit of code to match the names of the existing volumes and retrieve the data from the input state. Let me know if there is a better way to approach this, integration tests are passing, docs are added and a test is added for this along with an example.

**Comment by lawrencegripper** [comment]
Is there anything outstanding on me on this one atm?

File: azurerm/resource_arm_container_group.go [hunk 1]
```diff
@@ -185,13 +279,13 @@ func resourceArmContainerGroup() *schema.Resource {
 							Elem:     &schema.Schema{Type: schema.TypeString},
 						},
 
-						"volume": {
```
Comment (katbyte) [inline azurerm/resource_arm_container_group.go]: As this is a breaking change we are going to have to support both methods here until 2.0

File: azurerm/resource_arm_container_group_test.go [hunk 2]
```diff
@@ -512,6 +686,17 @@ resource "azurerm_container_group" "test" {
 	os_type             = "linux"
 	restart_policy      = "OnFailure"
 
+	volume {
+    name = "logs"
```
Comment (katbyte) [inline azurerm/resource_arm_container_group_test.go]: Could we property format this?

**Comment by lawrencegripper** [comment]
@katbyte thanks for the review, really appreciated. 

I'll try and find some time to fix up these bits, was originally just looking for an initial review of the approach before tidying things up so sorry about the messy bits!

Do you have any examples in the codebase of where an existing breaking change has been handled? That way I can review the approach and hopefully get it right.

**Pull request closed without merging**